sites/cnfpt/src/backend/base/tests/test_cache.py

```diff
@@ -0,0 +1,159 @@
+        logger logs the exception then fallback_cache takes over
+        """
+        client = RedisCacheWithFallback(None, {})
+        client.get("irrelevent")
+
+        redis_cache_mock.assert_called_once()
+        fallback_cache_mock.assert_not_called()
+
+        redis_cache_mock.side_effect = Exception()
+        client.get("irrelevent")
+
+        logger_mock.warning.assert_called_with(
+            "[DEGRADED CACHE MODE] - Switch to fallback cache"
+        )
+        logger_mock.exception.assert_called_once()
+        fallback_cache_mock.assert_called_once()
+
+    @override_settings(
+        CACHES={
+            "default": {
+                "BACKEND": "base.cache.RedisCacheWithFallback",
+                "LOCATION": "mymaster/redis-sentinel:26379,redis-sentinel:26379/0",
+                "OPTIONS": {
+                    "CLIENT_CLASS": "richie.apps.core.cache.SentinelClient",
+                },
+            },
+            "fallback": {
+                "BACKEND": "django.core.cache.backends.dummy.DummyCache",
+            },
+        }
+    )
+    @mock.patch.object(DummyCache, "delete")
+    @mock.patch.object(RedisCacheWithFallback, "_call_redis_cache")
+    def test_invalidate_fallback_cache(self, redis_cache_mock, delete_mock):
+        """
+        Test that fallback_cache is invalidated every 60 seconds
+        when Redis is up
+        """
+        now = datetime.datetime.now()
+
+        client = RedisCacheWithFallback(None, {})
+        client.get("round_0")
+        redis_cache_mock.reset_mock()
+        delete_mock.reset_mock()
+
+        # A second cache access should not delete the fallback cache again
+        client.get("round_1")
+        redis_cache_mock.assert_called_once()
+        delete_mock.assert_not_called()
+        redis_cache_mock.reset_mock()
+        delete_mock.reset_mock()
+
+        # Another call to Redis cache 120 seconds later
+        # should delete the fallback cache again
+        read_time = now + datetime.timedelta(seconds=120)
+        with mock.patch("base.utils.datetime") as mock_datetime:
+            mock_datetime.now.return_value = read_time
+            client.get("round_2")
+
+        redis_cache_mock.assert_called_once()
+        delete_mock.assert_called_once()
+        redis_cache_mock.reset_mock()
+        delete_mock.reset_mock()
+
+        # Another call to Redis cache 30 seconds later (<1 minute)
+        # should not delete the fallback cache again
+        read_time += datetime.timedelta(seconds=30)
+        with mock.patch("base.utils.datetime") as mock_datetime:
+            mock_datetime.now.return_value = read_time
+            client.get("round_3")
+
+        redis_cache_mock.assert_called_once()
+        delete_mock.assert_not_called()
+        redis_cache_mock.reset_mock()
+        delete_mock.reset_mock()
+
+        # Another call to Redis cache exactly 1 minute after
+        # the first call should not delete the fallback cache again
```

Comment:
Make it easier to understand the test, since it's actually the time elapsed since the latest call and not the first:
```suggestion
        # the latest call should not delete the fallback cache again
```